otelconf/v0.3.0/log_test.go

```diff
@@ -694,3 +704,217 @@ func TestLogProcessor(t *testing.T) {
 		})
 	}
 }
+
+func Test_otlpGRPCLogExporter(t *testing.T) {
+	type args struct {
+		ctx        context.Context
+		otlpConfig *OTLP
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    sdklog.Exporter
+		wantErr string
+	}{
+		{
+			name: "no TLS config",
+			args: args{
+				ctx: context.Background(),
+				otlpConfig: &OTLP{
+					Protocol:    ptr("grpc"),
+					Compression: ptr("gzip"),
+					Timeout:     ptr(5000),
+					Insecure:    ptr(true),
+					Headers: []NameStringValuePair{
+						{Name: "test", Value: ptr("test1")},
+					},
+				},
+			},
+			want: &otlploggrpc.Exporter{},
+		},
+		{
+			name: "with TLS config",
+			args: args{
+				ctx: context.Background(),
+				otlpConfig: &OTLP{
+					Protocol:    ptr("grpc"),
+					Compression: ptr("gzip"),
+					Timeout:     ptr(5000),
+					Certificate: ptr("testdata/server-certs/server.crt"),
+					Headers: []NameStringValuePair{
+						{Name: "test", Value: ptr("test1")},
+					},
+				},
+			},
+			want: &otlploggrpc.Exporter{},
+		},
+		{
+			name: "with TLS config and client key",
+			args: args{
+				ctx: context.Background(),
+				otlpConfig: &OTLP{
+					Protocol:          ptr("grpc"),
+					Compression:       ptr("gzip"),
+					Timeout:           ptr(5000),
+					Certificate:       ptr("testdata/server-certs/server.crt"),
+					ClientKey:         ptr("testdata/client-certs/client.key"),
+					ClientCertificate: ptr("testdata/client-certs/client.crt"),
+					Headers: []NameStringValuePair{
+						{Name: "test", Value: ptr("test1")},
+					},
+				},
+			},
+			want: &otlploggrpc.Exporter{},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			n, err := net.Listen("tcp", "localhost:0")
+			require.NoError(t, err)
+
+			tt.args.otlpConfig.Endpoint = ptr(strings.ReplaceAll(n.Addr().String(), "127.0.0.1", "localhost"))
+
+			tlsMode := ""
+			if tt.args.otlpConfig.Insecure == nil || !*tt.args.otlpConfig.Insecure {
+				tlsMode = "TLS"
+			}
+			if tt.args.otlpConfig.ClientCertificate != nil && *tt.args.otlpConfig.ClientCertificate != "" {
+				tlsMode = "mTLS"
+			}
+
+			col, err := newGRPCLogsCollector(n, tlsMode)
+			require.NoError(t, err)
+			defer col.srv.Stop()
+
+			exporter, err := otlpGRPCLogExporter(tt.args.ctx, tt.args.otlpConfig)
+			if tt.wantErr != "" {
+				require.Error(t, err)
+				require.Equal(t, tt.wantErr, err.Error())
+			} else {
```

Comment:
Isn't this unnecessary?

Reply:
yes, that was a copy-paste relict a previous test. i have removed that now